include/mango_v3.hpp

```diff
@@ -140,6 +142,39 @@ struct MangoAccount {
   uint8_t padding[5];
 };
 
+class MangoAccount {
+  // Creating a MangoAccount might involve non-trivial work like calling RPC
+  // methods, so we provide explict `MangoAccount::from()` overloaded set of
+  // functions and make the public constructor un-callable.
+  struct NotPubliclyConstructible {};
+
+ public:
+  MangoAccount(NotPubliclyConstructible /* npc */,
+               const MangoAccountInfo &accountInfo) noexcept
+      : accountInfo_(accountInfo) {}
+  // Use a prefetched accountInfo
+  static std::unique_ptr<MangoAccount> from(
+      const MangoAccountInfo &accountInfo) noexcept {
+    return std::make_unique<MangoAccount>(NotPubliclyConstructible{},
+                                          accountInfo);
+  }
+  // Fetch `accountInfo` from `endpoint` and decode it
+  static std::unique_ptr<MangoAccount> from(
+      const solana::PublicKey &pubKey,
+      const std::string &endpoint = MAINNET.endpoint) {
+    auto connection = solana::rpc::Connection(endpoint);
+    const auto &accountInfo =
+        connection.getAccountInfo<MangoAccountInfo>(pubKey.toBase58());
+    return std::make_unique<MangoAccount>(NotPubliclyConstructible{},
```

Comment:
unique_ptr with reference to rvalue? uuh i'm not sure this won't segfault

Reply:
also I try to call toBase58() in any internal method, strictly only in examples as it does come at considerable cost which is bad for latency, rather try to design the interfaces that the user needs to call it and hence can decide to cache the result.

Reply:
> unique_ptr with reference to rvalue?

Totally missed this, thanks for noticing it.

Reply:
Unless you need to use the heap, try not to introduce unique ptrs. Best left to the users to decide, when they want to work on stack and when on heap.

Reply:
> as it does come at considerable cost which is bad for latency

I will change the `PubKey` parameter to a `string` ,  just like `getAccountInfo` as that is what it is used for in this method really.